test/AutoDiff/SILOptimizer/closure_specialization.sil

```diff
@@ -0,0 +1,116 @@
+// RUN: %target-sil-opt -test-runner %s -o /dev/null 2>&1 | %FileCheck %s
```

Comment:
Will you please add original swift testcases? And similar ones to ensure these are handled.

Reply:
> Will you please add original swift testcases? And similar ones to ensure these are handled. You'd also need to ensure that generics (and corresponding reabstractions) are handled properly. There also might be index subset thunks, not 100% sure if these are not affected.

@asl What is the right way to add original Swift test-cases as XFAIL? 

I'm going to have to swap out the original closure-specialization optimization with the new swift based one in the optimization pipeline. But I can't think of a good way to let LIT know that the tests should be XFAIL when we're using the swift based closure-specialization optimization.

One option could be to simply copy all the test case files and mark them as XFAIL. But this does not seem viable.

@eeckstein do you have any thoughts on this?